### PR TITLE
Let CI use Rust 1.81

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

**Comment by shaobo-he-aws** [comment]
zigbuild issue: https://github.com/rust-cross/cargo-zigbuild/issues/289

File: .github/workflows/run_cedar_java_reusable.yml
Comment (john-h-kastner-aws) [inline .github/workflows/run_cedar_java_reusable.yml:1]: Could we use the `rust-toolchain.toml` file here like I did for the wasm build? That way we shouldn't have to remember to set the correct version when building for a release.

File: .github/workflows/run_cedar_java_reusable.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
           ref: ${{ inputs.cedar_policy_ref }}
           path: ./cedar
       - name: Prepare Rust Build
-        run: rustup install stable && rustup default stable 
+        run: rustup install 1.81 && rustup default 1.81 && rustup component add rustfmt
```
Comment (john-h-kastner-aws) [inline .github/workflows/run_cedar_java_reusable.yml:43]: Leave a comment to document the why we use 1.81

File: .github/workflows/run_cedar_java_reusable.yml
Comment (shaobo-he-aws) [reply .github/workflows/run_cedar_java_reusable.yml:1]: Can we approve this first so that other PRs are not blocked? I can work on it in another PR.

Comment (john-h-kastner-aws) [reply .github/workflows/run_cedar_java_reusable.yml:1]: sure

**Review by john-h-kastner-aws — APPROVED** [review]

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**